openml/extensions/sklearn/extension.py

```diff
@@ -769,21 +773,25 @@ def _get_external_version_string(
         external_versions.add(openml_version)
         external_versions.add(sklearn_version)
         for visitee in sub_components.values():
+            if isinstance(visitee, str):
+                continue
             for external_version in visitee.external_version.split(','):
                 external_versions.add(external_version)
         return ','.join(list(sorted(external_versions)))
 
     def _check_multiple_occurence_of_component_in_flow(
         self,
         model: Any,
-        sub_components: Dict[str, OpenMLFlow],
+        sub_components: Dict[str, Any],
```

Comment:
Hm, why can this map to `Any` now?

Reply:
The annotations used earlier expected the _subcomponent_ to be OpenML, however, we now need to handle _str_ too. I had the type annotations as `Dict[str, Union[str, OpenMLFlow]]`. However, flake8 was not able to decipher the rest of the if...else bodies and was raising potential errors, which I am handling with exactly those if...else cases. 
I agree that having *Any* is loosening the type checks but I wanted to see if these changes affect test cases on Travis, so I went ahead with Any in this draft PR since I was unable to think of a better solution.

Reply:
Could you please paste the error you got for the annotation `Dict[str, Union[str, OpenMLFlow]]`?

Reply:
There were a dozen or more :neutral_face: 
One of them, for instance, was for [this](https://github.com/openml/openml-python/blob/78b163e8a3f7fe2c977a26912f7a7a4ae2742f2a/openml/extensions/sklearn/extension.py#L700).
_str has got no 'name' attribute'_

Reply:
I think the solution is to use an if/elif/else statement here, where the access to `name` is after a check if the instance is of type `OpenMLFlow`. I'm not really happy about this as it bloat up the code, but it will allow us to keep the annotations as meaningful as possible.

Reply:
Though test cases passed locally, it might fail on travis since I still get these two flake issues.

`openml/extensions/sklearn/extension.py:698: error: Item "str" of "Union[OpenMLFlow, str, None]" has no attribute "name"`
`openml/extensions/sklearn/extension.py:698: error: Item "None" of "Union[OpenMLFlow, str, None]" has no attribute "name"`

I am unable to think of how I can bypass how flake is parsing the if-conditions.
It doesn't make sense, for multiple types.

Reply:
That's annoying. You could try [cast](https://docs.python.org/3/library/typing.html#typing.cast).

Reply:
This was fixed.